cpp/collections_module/algorithm/algorithm.cpp

```diff
@@ -0,0 +1,25 @@
+#include "algorithm.hpp"
+#include <unordered_set>
+
+void Collections::toSet(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+  try {
+    mgp::List list = arguments[0].ValueList();
+    std::unordered_set<mgp::Value, std::hash<mgp::Value>> set;
+    mgp::List return_list;
+    for (auto list_elem : list) {
+      if (set.find(list_elem) == set.end()) {
+        set.insert(list_elem);
+        return_list.AppendExtend(std::move(list_elem));
+      }
+    }
```

Comment:
And here you just go over the elements of set and move them to list

Reply:
FYI: when I do that, I messaged you on SLack, the order is not in reverse after all like I messaged, but its completely random, I guess thats okay because of the hashes.